test/active_support/cache/redis_store_test.rb

```diff
@@ -351,6 +351,34 @@ def setup
     end
   end
 
+  it "prevents cache stampede on fetch if race_condition_ttl is set" do
+    first_rabbit = second_rabbit = nil
+    options = { force: true, expires_in: 1.second, race_condition_ttl: 2.seconds }
+    @store.fetch("rabbit", options) { @rabbit }
+    sleep 1
+
+    th1 = Thread.new do
+      first_rabbit = @store.fetch("rabbit", race_condition_ttl: 2) do
+        sleep 1
+        @white_rabbit
+      end
+    end
+
+    sleep 0.1
```

Comment:
This is just to assure that first thread already bumped the stale record back to redis, but still hasn't finished generating new value. The other option is to send race_condition_ttl option to second thread as well (or just to have test for both cases).